cmd/sidecars/network-passt-binding/main.go

```diff
@@ -48,9 +48,15 @@ func main() {
 	defer os.Remove(socketPath)
 
 	server := grpc.NewServer([]grpc.ServerOption{}...)
-	hooksInfo.RegisterInfoServer(server, srv.InfoServer{Version: "v1alpha2"})
-	hooksV1alpha2.RegisterCallbacksServer(server, srv.V1alpha2Server{})
+	hooksInfo.RegisterInfoServer(server, srv.InfoServer{Version: "v1alpha3"})
+
+	log.Log.Infof("passt sidecar is now exposing its services on socket %s using %q API version", socketPath, "v1alpha3")
+	errChan := make(chan error)
+	go func() {
+		errChan <- server.Serve(socket)
```

Comment:
I think that `Serve` can do all of this internal.
e.g. pass to it the shutdown channel, then it will do the work and wait for the shutdown there.

Reply:
I understand I cause confusion.
I mean this:

```go
shutdown := make(chan struct{})
hooksV1alpha3.RegisterCallbacksServer(server, srv.V1alpha3Server{Done: shutdown})
srv.Serve(server, shutdown)
```

Reply:
done. I hope that's what you meant.